biocrnpyler/mechanisms_conformation.py

```diff
@@ -0,0 +1,37 @@
+
+# Copyright (c) 2020, Build-A-Cell. All rights reserved.
+# See LICENSE file in the project root directory for details.
+
+from .mechanism import Mechanism
+from .reaction import Reaction
+from .species import Complex, Species
+
+
+class One_Step_Reversible_Conformation_Change(Mechanism):
+    """A reaction where n binders (A) bind to 1 bindee (B) in one step
+       n A + B <--> nA:B
+    """
+    def __init__(self, name="one_step_conformation_change",
+                 mechanism_type="conformation_change"):
+        Mechanism.__init__(self, name, mechanism_type)
+
+    def update_species(self, s0, sf, additional_species = None, component = None, part_id = None, **kwords):
+
+        if additional_species is None:
+            additional_species = []
+
+        return [s0, sf] + additional_species
+
+
+    def update_reactions(self, s0, sf, additional_species = None, component = None, part_id = None, **kwords):
+
+        if part_id is None:
+            repr(s0)+"-"+repr(sf)
+
+        if additional_species is None:
+            additional_species = []
+
+        kf = component.get_parameter("kf", part_id = part_id, mechanism = self)
+        kr = component.get_parameter("kr", part_id = part_id, mechanism = self)
```

Comment:
so there is no default "k" to use for this? I am just wondering if i specify "kf" in parameters (nonspecifically) i guess that would cause a lot of problems

Reply:
kf = kr = 100 for instance would allow for fast transfer between different complexes. I also built in a way to rename the part_id for each conformation (self._get_part_id(s)) to make parameter files cleaner because the strings used to represent polymer conformations are gross.